### PR TITLE
goal: Filter all_installed for requested names

### PR DESCRIPTION
**Pull request by m-blaha**
We do not want to add all installed packages to upgrade job. Only installed packages with matching names are relevant.

https://bugzilla.redhat.com/show_bug.cgi?id=2142257

Test: https://github.com/rpm-software-management/ci-dnf-stack/pull/1180

### CODE REVIEW

File: libdnf/base/goal.cpp [hunk 1]
```diff
@@ -1164,6 +1164,7 @@ void Goal::Impl::add_up_down_distrosync_to_goal(
             // Otherwise it will also install not the latest installonly packages
             query.filter_available();
             all_installed.filter_latest_evr();
+            all_installed.filter_name(query);
```
Comment (j-mracek) [inline libdnf/base/goal.cpp:1167]: I think it is a correct solution. The code here is quite delicate therefore I recommend to enhance the description of the issue in commit message (description of reproducer can also help) plus add a reference to DNF5 bug (plus reference to dnf4 bug). I also suggest to enhance documentation in code. In DNF4, the code was modified just recently including a long description what and why it is there (base line 2171). May I ask you for the description enhancement?

Comment (m-blaha) [reply libdnf/base/goal.cpp:1167]: Yeah, sure. Will do.
Also I need to investigate why dnf5 with the issue has been passing ci tests for package upgrades - this should also result in a reproducer. Let me block the PR till I have a CI test for it.

Comment (j-mracek) [reply libdnf/base/goal.cpp:1167]: I think I have a reproducer. When I try to upgrade a leave package (dnf5) then it works like expected. When I upgrade package that will require upgrade of another installed package, the it writes nothing to do.

Comment (m-blaha) [reply libdnf/base/goal.cpp:1167]: Great, thanks!

Comment (m-blaha) [reply libdnf/base/goal.cpp:1167]: Uff, it took me a while but I finally have a reproducer. Not only the upgraded package must have a dependency, but also the installed versions must not be available in repos... I'm about to send a PR with ci test.

Comment (j-mracek) [reply libdnf/base/goal.cpp:1167]: Perfect

Comment (m-blaha) [reply libdnf/base/goal.cpp:1175]: Actually the absence in repos is not required - best=False is. I should have read the comment in dnf4 beforehand :(
Anyway, the test is ready and I've added required descriptions.

Comment (kontura) [reply libdnf/base/goal.cpp:1175]: I am sorry, I should have added the comment here as well.

**Review by kontura — APPROVED** [review]

**Comment by j-mracek** [comment]
LGTM

**Pull request merged**